pkg/cloud/vsphere/context/cluster_context.go

```diff
@@ -45,27 +46,29 @@ import (
 
 // ClusterContextParams are the parameters needed to create a ClusterContext.
 type ClusterContextParams struct {
-	Context    context.Context
-	Cluster    *clusterv1.Cluster
-	Client     client.ClusterV1alpha1Interface
-	CoreClient corev1.CoreV1Interface
-	Logger     logr.Logger
+	Context          context.Context
+	Cluster          *clusterv1.Cluster
+	Client           client.ClusterV1alpha1Interface
+	CoreClient       corev1.CoreV1Interface
+	ControllerClient controllerClient.Client
+	Logger           logr.Logger
 }
 
 // ClusterContext is a Go context used with a CAPI cluster.
 type ClusterContext struct {
 	context.Context
-	Cluster       *clusterv1.Cluster
-	ClusterCopy   *clusterv1.Cluster
-	ClusterClient client.ClusterInterface
-	ClusterConfig *v1alpha1.VsphereClusterProviderSpec
-	ClusterStatus *v1alpha1.VsphereClusterProviderStatus
-	CoreClient    corev1.CoreV1Interface
-	Logger        logr.Logger
-	client        client.ClusterV1alpha1Interface
-	machineClient client.MachineInterface
-	user          string
-	pass          string
+	Cluster          *clusterv1.Cluster
+	ClusterCopy      *clusterv1.Cluster
+	ClusterClient    client.ClusterInterface
+	ClusterConfig    *v1alpha1.VsphereClusterProviderSpec
+	ClusterStatus    *v1alpha1.VsphereClusterProviderStatus
+	CoreClient       corev1.CoreV1Interface
+	ControllerClient controllerClient.Client
+	Logger           logr.Logger
+	client           client.ClusterV1alpha1Interface
+	machineClient    client.MachineInterface
+	user             string
+	pass             string
```

Comment:
not your change, but the password in ClusterSpec should be encoded? i should user should pass encoded value instead of plain text password

Reply:
It never was, and if we want to discuss that, let’s file a different issue. I don’t believe it has any value as we can’t encrypt it since we’d need to supply the key, and base64 encoding it just obfuscates, not protects.

Reply:
Agree, if keeping plaintext credentials in memory is a problem let’s file an issue for that. Encoding to base64 is not helping in this case.

Reply:
Hi @frapposelli,

Yeah, and I only do this (@figo is incorrect, this *is* my code :) ) since the user and pass can come from a secret or the cluster spec. I can always pull them from the secret each time if I need to do so.

Reply:
Let's not forget, however, that we put plain-text credentials in a cluster spec for the initial target cluster since we cannot provide a secret to `clusterctl` in any easy manner today. It's not ideal, that's for sure.

Reply:
Let's discuss in a separate issue, it is not about this change, sorry for causing confsut